pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -236,6 +248,70 @@ func (pc *PackageControllerClient) Enable(ctx context.Context) error {
 	return nil
 }
 
+// UpdateSecrets is used to update the registry-mirror-cred secret used by the packages controller.
+func (pc *PackageControllerClient) UpdateSecrets(ctx context.Context, client client.Client, cluster *anywherev1.Cluster) error {
+	secretName := "registry-mirror-cred"
+	secretKey := types.NamespacedName{
+		Namespace: constants.EksaPackagesName,
+		Name:      secretName,
+	}
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: constants.EksaPackagesName,
+		},
+	}
+	credErr := client.Get(ctx, secretKey, secret)
+	err := fillRegistrySecret(cluster.Name, cluster.Spec.RegistryMirrorConfiguration, secret)
+	if err != nil {
+		return err
+	}
+
+	if apierrors.IsNotFound(credErr) {
+		return client.Create(ctx, secret)
+	} else if credErr == nil {
+		return client.Update(ctx, secret)
+	}
+	return credErr
+}
+
+func fillRegistrySecret(clusterName string, registry *anywherev1.RegistryMirrorConfiguration, secret *corev1.Secret) error {
```

Comment:
how was this secret created before?
I was expecting to have already code that did this. If it's helm, why don't we use helm to update the values as well? this would be duplicated logic in that case
It's possible that I'm not getting this right

Reply:
> it will use the image found in packages bundle instead where it rolls out a new pod. Once that pod rolls out with the new image, the secrets are reset to their default values.

why would the secret be reset to default after pod rolls out with image in package bundle? shouldn't we try solving this issue instead?